lib/strong_migrations/safe_methods.rb

```diff
@@ -45,15 +45,25 @@ def safe_add_reference(table, reference, *args, **options)
     end
 
     def safe_add_foreign_key(from_table, to_table, *args, **options)
+      validate_options = remove_options = options.slice(:column, :name)
       @migration.reversible do |dir|
         dir.up do
-          @migration.add_foreign_key(from_table, to_table, *args, **options.merge(validate: false))
+          # https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L1154C96-L1154C120
+          # if_not_exists does not check again `name` option.
```

Comment:
Until a fix is done/merged on Rails, I believe we still need to leave this.

This was detected by `AddForeignKeyName` test class, as it creates 2 foreign keys on the same column but with different names ( It's quite rare I think for users of this gem to encounter this case but nonetheless, it's more robust and makes tests pass )